### PR TITLE
2157  Tradelogs post processor service

### PR DESCRIPTION
**Pull request by halink0803**

### CODE REVIEW

File: tradelogs/cmd/trade-logs-post-processor/main.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+	influxClient, err := influxdb.NewClientFromContext(c)
+	if err != nil {
+		return err
+	}
+
+	sugar.Debugw("influx client initiate successfully", "influx client", influxClient)
+
+	// get first timestamp from db
+	q := fmt.Sprintf(`SELECT eth_volume FROM rsv_volume_day ORDER BY ASC LIMIT 1`)
```
Comment (favadi) [inline tradelogs/cmd/trade-logs-post-processor/main.go]: we should aggregate from trades measurement directly.

File: tradelogs/cmd/trade-logs-post-processor/main.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		}
+		res = response.Results
+	} else {
+		return res, err
+	}
+	return res, nil
+}
+
+func run(c *cli.Context) error {
+
```
Comment (favadi) [inline tradelogs/cmd/trade-logs-post-processor/main.go]: blank line.

File: tradelogs/cmd/trade-logs-post-processor/main.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	if err != nil {
+		return err
+	}
+
+	startTimeString := res[0].Series[0].Values[0][0].(string)
+	startTime, err := time.Parse(time.RFC3339, startTimeString)
+	if err != nil {
+		return err
+	}
+	// run each 10-day
```
Comment (favadi) [inline tradelogs/cmd/trade-logs-post-processor/main.go]: 1 day is enough? Have you try in dev database, how long does it take?

File: tradelogs/cmd/trade-logs-post-processor/main.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	for {
+		beginOfThisMonth := now.New(startTime).BeginningOfMonth()
+		previousMonth := beginOfThisMonth.Add(-24 * time.Hour)
+		beginOfLastMonth := now.New(previousMonth).BeginningOfMonth()
+		query := fmt.Sprintf(`SELECT SUM(eth_volume) AS eth_volume, SUM(usd_volume) AS usd_volume
+		INTO monthly_reserve_volume FROM rsv_volume_day WHERE time >= '%s' AND time < '%s'
+		GROUP BY src_rsv_addr, dst_rsv_addr`, beginOfLastMonth.Format(time.RFC3339), beginOfThisMonth.Format(time.RFC3339))
```
Comment (favadi) [inline tradelogs/cmd/trade-logs-post-processor/main.go]: As discussed, we need all available fields in the summary report. Please update.

File: tradelogs/cmd/trade-logs-post-processor/main.go [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+		return err
+	}
+	// run each day
+	for {
+		beginOfThisMonth := now.New(startTime).BeginningOfMonth()
+		previousMonth := beginOfThisMonth.Add(-24 * time.Hour)
+		beginOfLastMonth := now.New(previousMonth).BeginningOfMonth()
+
+		// reserve volume monthly for dst reserve
+		query := fmt.Sprintf(`SELECT SUM(eth_amount) AS eth_volume, SUM(usd_amount) AS usd_volume FROM
```
Comment (cyberliem) [inline tradelogs/cmd/trade-logs-post-processor/main.go]: Please exclude where src_rsv_addr =''. It's a waste of resource to query into memory and exclude it later. I'm doing some test to see if it's wrong when we don't filter src_rsv_addr. 
Please exclude WETH-ETH pair as well.

File: tradelogs/cmd/trade-logs-post-processor/main.go [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+	// run each day
+	for {
+		beginOfThisMonth := now.New(startTime).BeginningOfMonth()
+		previousMonth := beginOfThisMonth.Add(-24 * time.Hour)
+		beginOfLastMonth := now.New(previousMonth).BeginningOfMonth()
+
+		// reserve volume monthly for dst reserve
+		query := fmt.Sprintf(`SELECT SUM(eth_amount) AS eth_volume, SUM(usd_amount) AS usd_volume FROM 
+		(SELECT eth_amount, eth_amount*eth_usd_rate as usd_amount FROM trades WHERE time >= '%s' AND time < '%s'
+		GROUP BY src_rsv_addr) GROUP BY src_rsv_addr`, beginOfLastMonth.Format(time.RFC3339), beginOfThisMonth.Format(time.RFC3339))
```
Comment (cyberliem) [inline tradelogs/cmd/trade-logs-post-processor/main.go]: should use time UTC() before Format.

Comment (halink0803) [reply tradelogs/cmd/trade-logs-post-processor/main.go]: Timezone does not affect here.

File: tradelogs/cmd/trade-logs-post-processor/main.go (hunk 5 above)
Comment (halink0803) [reply tradelogs/cmd/trade-logs-post-processor/main.go]: Excluded.

File: tradelogs/cmd/trade-logs-post-processor/main.go [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+		sugar.Debug("query ", query)
+
+		_, err = queryDB(influxClient, query)
+		if err != nil {
+			return err
+		}
+
+		if beginOfThisMonth.Equal(now.New(time.Now().In(time.UTC)).BeginningOfMonth()) {
+			sugar.Info("Sleep for 10 day...")
+			time.Sleep(24 * time.Hour)
```
Comment (favadi) [inline tradelogs/cmd/trade-logs-post-processor/main.go]: the log states that sleep for 10 day, but we only sleep for 24 hour,

Comment (favadi) [reply tradelogs/cmd/trade-logs-post-processor/main.go]: Anw, please just remove the sleep logic, we will use a timer to run it daily.

File: tradelogs/cmd/trade-logs-post-processor/main.go [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+	// run each day
+	for {
+		beginOfThisMonth := now.New(startTime).BeginningOfMonth()
+		previousMonth := beginOfThisMonth.Add(-24 * time.Hour)
+		beginOfLastMonth := now.New(previousMonth).BeginningOfMonth()
+
+		// reserve volume monthly for dst reserve
+		query := fmt.Sprintf(`SELECT SUM(eth_amount) AS eth_volume, SUM(usd_amount) AS usd_volume FROM 
+		(SELECT eth_amount, eth_amount*eth_usd_rate as usd_amount FROM trades WHERE time >= '%s' AND time < '%s' AND src_rsv_addr != '' 
+		AND ((src_addr != '%s' AND dst_addr != '%s')
```
[the remark this answers is not in the document]
Comment (halink0803) [reply tradelogs/cmd/trade-logs-post-processor/main.go]: @cyberliem  Fixed this issue but issue still exist in volume_cqs. Please check redmine and remember to not expose redmine endpoint as it is private. I edited your comment and hide the sensitive content.

**Pull request merged**